proposed/0006-Launcher.md

```diff
@@ -0,0 +1,204 @@
+
+**step 7:** Launcher Background Process will listen on local host: `127.0.9.9:30000`. If unavailable, it will increment the port till a valid TCP Listener is in place. This will be thus `127.0.9.9:Launcher-Port`. This will remain on till either the OS is shutdown or Launcher Background Process is killed. We will call this combination `<Laucher-IP:Launcher-Port>`.
+
+## Add App Flow
+
+**step 0:** User Drags `XYZ` App binary into the Launcher to add it.
+
+**step 1:** Launcher Creates (if it’s the first time it saw this App):
+- Unique Random 64 Byte Id for this app - App-ID (because names of different binaries can be same if they are in different locations on the same machine - thus we need a unique identifier for the binary across all machines)
+- Unique Directory Id and an associated Unique Root Directory Listing `<APP-ROOT-DIR>` for this app - `XYZ-Root-Dir`. For directory name conflicts append numbers so that they can exist on a file system - eg., `XYZ-1-Root-Dir`. This shall be created inside `<USER’S-PRIVATE-ROOT-DIRECTORY-ID>/`.
+- `<APP-ROOT-DIR>` shall be always **unversioned** and **private** (ie., encrypted with App-specific crypto keys). Any requirement for a versioned or public directory from the App can be managed by the App itself by creating further subdirectories.
+- The user is asked if it's wanted to share the access of SAFEDrive directory with this App.
+- Append this information in `<LAUNCHER-CONFIG-FILE>` = `<MAIDSAFE-SPECIFIC-CONFIG-ROOT>/LauncherReservedDirectory/LauncherConfigurationFile` (This is what the DNS crate currently does too inside `<MAIDSAFE-SPECIFIC-CONFIG-ROOT>/DnsReservedDirectory/DnsConfigurationFile`.) - Format shall be CBOR (compact-binary-object-representation).
+```
+[
+    {
+        App Name,
+        Reference Count,
+        Random Unique App ID,    // 64 Bytes
+        Vec<SHA512(App-Binary)>, // Also serves to tell how many machines where the App is added to Launcher
+        <APP-ROOT-DIR-Key>,
+        Option<SAFEDrive-Directory-Key>,
+        OtherMetadata, // For Future Use
+    }
+    {
+        App Name,
+        Reference Count,
+        Random Unique App ID,    // 64 Bytes
+        Vec<SHA512(App-Binary)>, // Also serves to tell how many machines where the App is added to Launcher
+        <APP-ROOT-DIR-Key>,
+        Option<SAFEDrive-Directory-Key>,
+        OtherMetadata, // For Future Use
+    },
+    … etc
+]
+```
+- Create/Append to local config file (on the user's machine) the following:
+```
+[
+    { App-ID, “/path/to/XYZ” }, // This is newly added by the above
+    { App-ID, “/path/to/some/other/app/already/there” },
+    ... etc.,
+]
+```
+- The format of the config file will be CBOR (compact-binary-object-representation). The name of the local config file should be `<LOCAL-CONFIG-FILE> = launcher.config`. The config file location flowchart shall be same as that of `crust` crate's.
```

Comment:
Maybe link to docs for this: https://github.com/maidsafe/crust/blob/03d63b4526cf87667c1d1775377398875072c12a/docs/vault_config_file_flowchart.pdf